Engine/DataFeeds/ZipDataCacheProvider.cs

```diff
@@ -132,7 +126,25 @@ public Stream Fetch(string key)
         /// <param name="data">The data as a byte array</param>
         public void Store(string key, byte[] data)
         {
-            //
+            DataCacheProviderExtensions.ParseKey(key, out var fileName, out var entryName);
+
+            // We only support writing to zips with this provider
+            if (fileName.EndsWith(".zip", StringComparison.InvariantCulture))
+            {
+                // If its not in the cache, and can not be cached we need to create it
+                if (!_zipFileCache.TryGetValue(fileName, out var cachedZip) && !Cache(fileName, out cachedZip))
+                {
+                    // Create the zip, if successful cache it for later use
+                    if (Compression.ZipData(fileName, entryName, data))
+                    {
+                        Cache(fileName, out _);
+                    }
+
+                    return;
+                }
+
+                cachedZip.WriteEntry(entryName, data);
```

Comment:
hm this method isn't really thread safe, same for `WriteEntry` bellow. Image the cases where the timer kicks in while it's being written too? or if another thread is ready while writing to it?